compiler/rustc_data_structures/src/sync.rs

```diff
@@ -377,51 +377,92 @@ impl<K: Eq + Hash, V: Eq, S: BuildHasher> HashMapExt<K, V> for HashMap<K, V, S>
     }
 }
 
+#[cfg(not(parallel_compiler))]
+static mut SHARED: bool = false;
+#[cfg(parallel_compiler)]
+static mut SHARED: bool = true;
```

Comment:
Can you use the unstable `SyncUnsafeCell<T>` instead of `static mut` for both versions? This would be a great place to test it out.

Reply:
is it ever mutated ?

Reply:
not for now, but I think we should change it based on `threads` option in a early stage of compilation

Reply:
We use `SyncUnsafeCell<T>` now